fasm2bels/models/gtp_channel_models.py

```diff
@@ -0,0 +1,158 @@
+import re
+import json
+import os
+from .verilog_modeling import Bel, Site, make_inverter_path
+
+ZERO_VAL_PARAMS = {
+    "CBCC_DATA_SOURCE_SEL": "ENCODED",
+    "RXBUF_ADDR_MODE": "FULL",
+    "RXOOB_CLK_CFG": "PMA",
+    "RXSLIDE_MODE": "OFF",
+    "RX_XCLK_SEL": "RXREC",
+    "SATA_PLL_CFG": "VCO_3000MHZ",
+    "TXPI_PPMCLK_SEL": "TXUSRCLK",
+    "TX_DRIVE_MODE": "DIRECT",
+    "TX_XCLK_SEL": "TXOUT",
+}
+
+
+def get_gtp_channel_site(db, grid, tile, site):
+    """ Return the prjxray.tile.Site object for the given GTP site. """
+    gridinfo = grid.gridinfo_at_tilename(tile)
+    tile_type = db.get_tile_type(gridinfo.tile_type)
+
+    sites = list(tile_type.get_instance_sites(gridinfo))
+
+    for site in sites:
+        if "GTPE2_CHANNEL" in site:
+            return site
+
+    assert False, (tile, site)
+
+
+def process_gtp_channel(conn, top, tile_name, features):
+    """
+    Processes the GTP_CHANNEL tile
+    """
+
+    # Filter only GTPE2_CHANNEL related features
+    gtp_channel_features = [
+        f for f in features if 'GTPE2_CHANNEL.' in f.feature
+    ]
+    if len(gtp_channel_features) == 0:
+        return
+
+    site = get_gtp_channel_site(
+        top.db, top.grid, tile=tile_name, site='GTPE2_CHANNEL')
+
+    # Create the site
+    gtp_site = Site(gtp_channel_features, site)
+
+    # Create the GTPE2_CHANNEL bel and add its ports
+    gtp = Bel('GTPE2_CHANNEL')
+    gtp.set_bel('GTPE2_CHANNEL')
+
+    # If the GTPE2_CHANNEL is not used then skip the rest
+    if not gtp_site.has_feature("IN_USE"):
+        return
+
+    db_root = top.db.db_root
```

Comment:
I think lines 59-95 are generic and not GTP specific.  Please refactor into a nice library function with good documentation?

Reply:
Sure, makes sense, will do

Reply:
Common code exported in utils